library/kani_core/src/arbitrary.rs

```diff
@@ -193,6 +193,87 @@ macro_rules! generate_arbitrary {
         pub mod slice {
             kani_core::slice_generator!();
         }
+
+        mod range_structures {
+            use super::{
+                Arbitrary,
+                core_path::{
+                    mem,
+                    ops::{Bound, Range, RangeFrom, RangeInclusive, RangeTo, RangeToInclusive},
+                },
+            };
+
+            impl<T> Arbitrary for Bound<T>
+            where
+                T: Arbitrary,
+            {
+                fn any() -> Self {
+                    match u8::any() % 3 {
+                        0 => Bound::Included(T::any()),
+                        1 => Bound::Excluded(T::any()),
+                        _ => Bound::Unbounded,
+                    }
+                }
+            }
+
+            impl<T> Arbitrary for Range<T>
+            where
+                T: Arbitrary + PartialOrd,
+            {
+                fn any() -> Self {
+                    let (mut first, mut second) = (T::any(), T::any());
+                    adjust(&mut first, &mut second);
+                    first..second
+                }
+            }
+
+            impl<T> Arbitrary for RangeFrom<T>
+            where
+                T: Arbitrary,
+            {
+                fn any() -> Self {
+                    T::any()..
+                }
+            }
+
+            impl<T> Arbitrary for RangeInclusive<T>
+            where
+                T: Arbitrary + PartialOrd,
+            {
+                fn any() -> Self {
+                    let (mut first, mut second) = (T::any(), T::any());
+                    adjust(&mut first, &mut second);
+                    first..=second
+                }
+            }
+
+            impl<T> Arbitrary for RangeTo<T>
+            where
+                T: Arbitrary,
+            {
+                fn any() -> Self {
+                    ..T::any()
```

Comment:
Why not create it via instantiating the object, e.g. `RangeTo { end: kani::any() }`?

Reply:
The individuals behind the language dedicated their time to created a special syntax for such scenario, why not take advantage of it? Ranges also have divergent initializations, for example, see `RangeInclusive`.

Regardless, I can instantiate them manually if you want.